jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -550,6 +575,13 @@ ecma_utf8_string_to_number (const lit_utf8_byte_t *str_p, /**< utf-8 string */
       {
         digit_value = (*str_p - LIT_CHAR_0);
       }
+#if ENABLED (JERRY_ESNEXT)
+      else if (*str_p == LIT_CHAR_UNDERSCORE && options)
```

Comment:
`(options & ECMA_CONVERSION_ALLOW_UNDERSCORE)`
Options is a flag set not a boolean.